lib/spack/llnl/util/filesystem.py

```diff
@@ -1068,6 +1068,16 @@ def working_dir(dirname: str, *, create: bool = False):
         os.chdir(orig_dir)
 
 
+@system_path_filter
+def safe_make_new_dir(path):
+    while True:
+        try:
+            os.mkdir(path)
+            break
+        except FileExistsError:
+            pass
```

Comment:
I was thinking this will need to try creating different pathnames, like if someone asks for "foo", we try "foo0", "foo1", "foo2"...

Otherwise, if the requested directory exists (i.e. some other tool within the Package's stage happens to create a directory of the same name) then this would loop forever

Reply:
Actually https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp might be what I was hoping for, so perhaps that would suffice here?

Reply:
Didn't realize you could use `mkstemp` in arbitrary directories, that sounds like a good plan!